develop/tutorials/articles/194-adaptive-media/00-adaptive-media-images-rest-apis-intro.markdown

```diff
@@ -1,312 +1,365 @@
+Adaptive Media images.
+
+You can learn how to use the */configuration* endpoint next.
+
+## Configuration [](id=configuration)
+
+The following requests are possible through the */configuration* endpoint:
+
+### List All Available Configurations [](id=list-all-available-configurations)
+
+Below is an example request pattern:
+
+    GET [server address]/o/adaptive-media-rest/images/configuration
+
+**Status Codes**
+
+200<br />
+500
+
+Below is an example response object:
+
+    [
+    {
+    "name": "Extra large demo size",
+    "id": "demo-xlarge",
+    "max-width": "1200",
+    "max-height": "1200"
+    },
+    {
+    "name": "Extra small demo size",
+    "id": "demo-xsmall",
+    "max-width": "50",
+    "max-height": "50"
+    },
+    {
+    "name": "Large demo size",
+    "id": "demo-large",
+    "max-width": "800",
+    "max-height": "800"
+    }
+    ]
+
+### Get a Configuration by its ID [](id=get-a-configuration-by-its-id)
+
+Below is an example request pattern:
+
+    GET [server address]/o/adaptive-media-rest/images/[configurationId]
+
+**Status Codes**
+
+200<br />
+404 (Non-existent configuration)
+
+Below is an example response object:
+
+    {
+    "name": "Small demo size",
+    "id": "demo-small",
+    "max-width": "100",
+    "max-height": "100"
+    }
+
+### Add a New Configuration [](id=add-a-new-configuration)
+
+Below is an example request pattern:
+
+    PUT [server address]/o/adaptive-media-rest/images/[configurationId]
+
+**Status Codes**
+
+200<br />
+400 (wrong body)<br />
+403 (user without permission)
+
+Below is an example request:
+
+    PUT [server address]/o/adaptive-media-rest/images/small
+
+    BODY:
+    {
+    "name": "Small Size",
+    "max-height": 100,
+    "max-width": 100
+    }
+
+Here is the resulting configuration:
+
+    {
+    "name": "Small Size",
+    "id": "small",
+    "max-height": "100",
+    "max-width": "100"
+    }
+
+### Delete a Configuration [](id=delete-a-configuration)
+
+Below is an example request pattern:
+
+    DELETE [server address]/o/adaptive-media-rest/images/[configurationId]
+
+**Status Codes**
+
+200<br />
+403 (user without permission)
+
+Below is an example request:
+
+    DELETE [server address]/o/adaptive-media-rest/images/small
+
+    BODY:
+    {
+    "name": "Small Size",
+    "max-height": 100,
+    "max-width": 100
+    }
+
+Here is the resulting deleted configuration:
+
+    {
+    "name": "Small Size",
+    "id": "small",
+    "max-height": "100",
+    "max-width": "100"
+    }
+
+Now that you know how to use the */configuration* endpoint, you can learn how to
+use the */content* endpoint next.
+
+## Content Endpoint [](id=content-endpoint)
+
+To use the */content* endpoint you must first choose the Adaptive Media
+image you want to recover. You can specify an adaptive media image via two
+possible forms of ID: `fileEntryId` or `fileVersionId`.
+<!-- Where can I find this information for the Adaptive Media images? -->
```

Comment:
Let's imagine we are building a mobile app with a list of images from our server. We want to save as many data as possible, so we will be using adaptive media to perform the requests.

All requests to JSON-WS are made to **_localhost:8080/api/jsonws_** or **_yourhost.com/api/jsonws_**

First you will retrieve the folder from where you have your pictures using: **_/dlapp/get-folder_**. This endpoint needs **_repositoryId_** (**_siteId_** in this case), **_parentFolderId_** (0 if the folder is in the root of DL) and the folder's name (for example: "Adaptive Media"). This will be the output:

```
{
    "companyId": "20116",
    "createDate": 1488272055498,
    "defaultFileEntryTypeId": "0",
    "description": "",
    "folderId": "37562",
    "groupId": "20143",
    "hidden": false,
    "lastPostDate": 1488272064826,
    "lastPublishDate": null,
    "modifiedDate": 1488272055498,
    "mountPoint": false,
    "name": "Adaptive Media",
    "parentFolderId": "0",
    "repositoryId": "20143",
    "restrictionType": 0,
    "status": 0,
    "statusByUserId": "0",
    "statusByUserName": "",
    "statusDate": null,
    "treePath": "/37562/",
    "userId": "37501",
    "userName": "Alejandro Hernandez",
    "uuid": "8cd2d842-295c-e8c9-eb25-1b844a725888"
}
```

From all this JSON, we'll use the **folderId**.

With that, we'll have to call **_/dlapp/get-file-entries_**. This endpoint will request: **_repositoryId_** (same as previous request), **_folderId_** (the folderId we have saved) and **_mimeTypes_** (["image/jpeg"], for example).

This will be the output:

```
[
  {
    "classNameId": "0",
    "classPK": "0",
    "companyId": "20116",
    "createDate": 1488272064826,
    "custom1ImageId": "0",
    "custom2ImageId": "0",
    "description": "",
    "extension": "jpeg",
    "extraSettings": "",
    "fileEntryId": "37611",
    "fileEntryTypeId": "0",
    "fileName": "27ecb9b1-b43d-4649-804e-d5af8b42f680.jpeg",
    "folderId": "37562",
    "groupId": "20143",
    "largeImageId": "0",
    "lastPublishDate": null,
    "manualCheckInRequired": false,
    "mimeType": "image/jpeg",
    "modifiedDate": 1488272064827,
    "name": "310",
    "readCount": 0,
    "repositoryId": "20143",
    "size": "572056",
    "smallImageId": "0",
    "title": "27ecb9b1-b43d-4649-804e-d5af8b42f680.jpeg",
    "treePath": "/37562/",
    "userId": "37501",
    "userName": "Alejandro Hernandez",
    "uuid": "2931c580-8323-046c-0fbb-9c0a729308a0",
    "version": "1.0"
  },
  {
    "classNameId": "0",
    "classPK": "0",
    "companyId": "20116",
    "createDate": 1488272063682,
    "custom1ImageId": "0",
    "custom2ImageId": "0",
    "description": "",
    "extension": "jpeg",
    "extraSettings": "",
    "fileEntryId": "37600",
    "fileEntryTypeId": "0",
    "fileName": "74dc5b63-d040-4db6-8e5d-ea66631fccfe.jpeg",
    "folderId": "37562",
    "groupId": "20143",
    "largeImageId": "0",
    "lastPublishDate": null,
    "manualCheckInRequired": false,
    "mimeType": "image/jpeg",
    "modifiedDate": 1488272063684,
    "name": "309",
    "readCount": 0,
    "repositoryId": "20143",
    "size": "860822",
    "smallImageId": "0",
    "title": "74dc5b63-d040-4db6-8e5d-ea66631fccfe.jpeg",
    "treePath": "/37562/",
    "userId": "37501",
    "userName": "Alejandro Hernandez",
    "uuid": "9d68875c-ad8f-5779-b5e7-3096019aeb61",
    "version": "1.0"
  }
]
```

From each of the json objects in this array we can extract the **_fileEntryId_** to be used in our AM endpoints.

For example, if we have created a configuration for mobile, with id _mobile_ we will perform a request like this for each image:

**_http://localhost:8080/o/adaptive-media-rest/images/content/file/37611/version/last/config/mobile_**

**_http://localhost:8080/o/adaptive-media-rest/images/content/file/37600/version/last/config/mobile_**

And we will receive the adaptive media mobile version of our images (or the original image in case the variants hasn't been generated yet)